astropy/convolution/convolve.py

```diff
@@ -1033,7 +1076,20 @@ def convolve_models(model, kernel, mode="convolve_fft", **kwargs):
     else:
         raise ValueError(f"Mode {mode} is not supported.")
 
-    return CompoundModel(operator, model, kernel)
+    if bounding_box is None or resolution is None:
+        none_kws = " and ".join(
+            [
+                k
+                for k, v in {
+                    "bounding_box": bounding_box,
+                    "resolution": resolution,
+                }.items()
+                if v is None
+            ]
+        )
+        raise ValueError(f"Required keyword(s) {none_kws} not provided or set to None")
+
+    return Convolution(operator, model, kernel, bounding_box, resolution, cache)
```

Comment:
An additional implicit breaking change here is that this path requires `scipy`, which isn't a required dependency to `astropy` (there's even a 2 years old issue proposing to change that #12448). Again, this could be okay to do in a major release (7.0 is the next one), but it would be a lot more gentle to users if we could warn about it ahead of time, and keep the code working as it does now until then. You'd need to check at runtime whether `scipy` is installed (use `astropy.utils.compat.HAS_SCIPY`) and emit about it being required in the future if it's missing.